cpp/memilio/utils/type_chart.h

```diff
@@ -0,0 +1,91 @@
+/* 
+* Copyright (C) 2020-2023 German Aerospace Center (DLR-SC)
+*
+* Authors: René Schmieding
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef MIO_UTILS_TYPE_CHART_H_
+#define MIO_UTILS_TYPE_CHART_H_
+
+#include "memilio/io/io.h"
+#include "memilio/utils/index.h"
+
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+namespace mio
+{
+
+/// @brief Collection of types. Each type is mapped to an index of type size_t.
+template <class... Types>
+class TypeChart
```

Comment:
It used to be called "FlowChart", but since it lost the direct connection to the type "Flows", we renamed it. I still think `TypeList` would be a more appropriate name.